components/org.wso2.carbon.identity.recovery.endpoint/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java

```diff
@@ -81,6 +81,12 @@ public Response recoverPasswordPost(RecoveryInitiatingRequestDTO recoveryInitiat
         } catch (IdentityRecoveryException e) {
             RecoveryUtil.handleInternalServerError(Constants.SERVER_ERROR, e.getErrorCode(), LOG, e);
         } catch (Throwable throwable) {
+            if (throwable != null && StringUtils.equals(Constants.ERROR_MESSAGE_EMAIL_NOT_FOUND,
+                    throwable.getMessage())) {
+                String msg = "You have not set the users email address. Please set the email address and reset again.";
+                LOG.error(msg);
+                RecoveryUtil.handleBadRequest(msg, Constants.ERROR_CODE_EMAIL_NOT_FOUND);
```

Comment:
```suggestion
                RecoveryUtil.handleBadRequest(throwable.getMessage(), Constants.ERROR_CODE_EMAIL_NOT_FOUND);
```

Reply:
Fixed in https://github.com/Geesa-Vihara/identity-governance/commit/c153383f87b8a1de4298f5a5c9526c2dab5e297a